Unable to bypass autoCompile mechanism

First of all I'm sorry for deleting the bug report template.
I did it because I have a demo project here: https://github.com/anotheruser584/wdio-ts-node-issue

What I'm trying to achieve is bypass WDIO auto-compile process. Or in other words I want to handle `ts-node` configuration manually the way it was possible in v6 (I believe).

The idea is to register `ts-node` in `entrypoint.js` and then feed `entrypoint.js` to WDIO. Meanwhile I have disabled `autoComplie` in `wdio.conf.ts`

From what I can see (I might be mistaken here) is that `ts-node` is registered twice. Again I might be wrong but you usually see this kind of verbose errors when `ts-node` is trying to compile already compiled `ts` file.
```
t.js: ⨯ Unable to compile TypeScript:
[0-0] configs/hooks/index.ts(2,5): error TS7022: '__createBinding' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
[0-0] configs/hooks/index.ts(2,58): error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
[0-0] configs/hooks/index.ts(2,84): error TS7006: Parameter 'o' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(2,87): error TS7006: Parameter 'm' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(2,90): error TS7006: Parameter 'k' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(2,93): error TS7006: Parameter 'k2' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(5,16): error TS7006: Parameter 'o' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(5,19): error TS7006: Parameter 'm' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(5,22): error TS7006: Parameter 'k' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(5,25): error TS7006: Parameter 'k2' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(9,5): error TS7022: '__exportStar' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
[0-0] configs/hooks/index.ts(9,60): error TS7006: Parameter 'm' implicitly has an 'any' type.
[0-0] configs/hooks/index.ts(9,63): error TS7006: Parameter 'exports' implicitly has an 'any' type.
[0-0]
[0-0] 2021-05-27T08:15:27.097Z ERROR @wdio/local-runner: Failed launching test session: TypeError: Cannot read property 'emit' of undefined
```

The issue above can be easily fixed by removing `require('ts-node')` in `entrypoint.js` and enabling `autoCompile`. In this case I can just remove `entrypoint.js` and change my script to `wdio wdio.conf.ts`.

Don't take me wrong but I really don't want to rely on WDIO internal blackbox where it is scanning dependencies. Also I don't want to pass `ts-node` options via WDIO config because it is unnecessary wrapper which, again makes me rely on WDIO internal mechanism.

--- a/packages/wdio-cli/src/launcher.ts
+++ b/packages/wdio-cli/src/launcher.ts
@@ -58,20 +58,10 @@ class Launcher {
         this.configParser = new ConfigParser()
 
         /**
-         * autocompile before parsing configs so we support ES6 features in configs, only if
+         * merge auto compile opts to understand how to parse the config
          */
-        if (
-            /**
-             * the auto compile option is not define in this case we automatically compile
-             */
-            typeof _args.autoCompileOpts?.autoCompile === 'undefined' ||
-            /**
-             * or it was define and its value is not false
-             */
-            (_args.autoCompileOpts?.autoCompile as any as string) !== 'false'
-        ) {
-            this.configParser.autoCompile()
-        }
+        this.configParser.merge({ autoCompileOpts: _args.autoCompileOpts })
+        this.configParser.autoCompile()
 
         this.configParser.addConfigFile(_configFilePath)
         this.configParser.merge(_args)

--- a/packages/wdio-cli/tests/launcher.test.ts
+++ b/packages/wdio-cli/tests/launcher.test.ts
@@ -35,12 +35,16 @@ describe('launcher', () => {
         it('should not run auto compile if cli param was provided', () => {
             const otherLauncher = new Launcher('./', {
                 autoCompileOpts: {
-                    // @ts-expect-error cli params are always strings
-                    autoCompile: 'false'
+                    autoCompile: false
                 }
             })
 
-            expect(otherLauncher['configParser'].autoCompile).toBeCalledTimes(0)
+            expect(otherLauncher['configParser'].merge).toBeCalledWith({
+                autoCompileOpts: {
+                    autoCompile: false
+                }
+            })
+            expect(otherLauncher['configParser'].autoCompile).toBeCalledTimes(1)
         })
     })
 

--- a/website/docs/TypeScript.md
+++ b/website/docs/TypeScript.md
@@ -37,6 +37,25 @@ export const config = {
 }
 ```
 
+If you don't want to use WebdriverIO's internal transpiler functionality you can create your own `entrypoint.js` file where `ts-node` is defined manually:
+
+```ts title="entrypoint.js"
+require('ts-node').register(
+    {
+        transpileOnly: false,
+        files: true,
+        project: "./tsconfig.json"
+    }
+)
+module.exports = require('./configs/wdio.conf')
+```
+
+In this case you have to pass `--no-autoCompileOpts.autoCompile` as parameter to the `wdio` command to disable auto compiling, e.g.:
+
+```sh
+npx wdio run ./entrypoint.js --no-autoCompileOpts.autoCompile
+```
+
 ## Framework Setup
 
 And your `tsconfig.json` needs the following:

